src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-pricing-page/ubs-admin-tariffs-pricing-page.component.ts

```diff
@@ -156,52 +154,72 @@ export class UbsAdminTariffsPricingPageComponent implements OnInit, OnDestroy {
     const { minPriceOfOrder, maxPriceOfOrder, minAmountOfBigBags, maxAmountOfBigBags, limitDescription } = this.limitsForm.value;
 
     const tariffId = this.selectedCardId;
-    const locationId = this.locationId;
-
-    this.bagInfo = {
-      min: minAmountOfBigBags,
-      max: maxAmountOfBigBags,
-      locationId
+    this.getCheckBoxStatus();
+
+    let limit: ILimit = {
+      bagLimitDtoList: this.getCheckBoxInfo(),
+      courierLimit: this.limitStatus,
+      limitDescription,
+      min: null,
+      max: null
     };
 
-    this.sumInfo = {
-      min: minPriceOfOrder,
-      max: maxPriceOfOrder,
-      locationId
-    };
+    if (this.limitStatus === limitStatus.limitByAmountOfBag) {
+      limit = { ...limit, min: minAmountOfBigBags, max: maxAmountOfBigBags };
+    } else {
+      limit = { ...limit, min: minPriceOfOrder, max: maxPriceOfOrder };
+    }
 
-    this.descriptionInfo = {
-      limitDescription
-    };
+    if (this.areAllCheckBoxEmpty) {
+      limit.min = null;
+      limit.max = null;
 
-    if (this.limitStatus === limitStatus.limitByPriceOfOrder) {
       this.tariffsService
-        .setLimitsBySumOrder(this.sumInfo, tariffId)
+        .setTariffLimits(limit, tariffId)
         .pipe(takeUntil(this.destroy))
         .subscribe(() => {
           this.getCouriers();
         });
-      this.changeDescription();
-    }
-
-    if (this.limitStatus === limitStatus.limitByAmountOfBag) {
+    } else {
       this.tariffsService
-        .setLimitsByAmountOfBags(this.bagInfo, tariffId)
+        .setTariffLimits(limit, tariffId)
         .pipe(takeUntil(this.destroy))
         .subscribe(() => {
           this.getCouriers();
         });
-      this.changeDescription();
-    }
-
-    if (this.limitStatus === null) {
-      this.changeDescription();
     }
 
     this.saveBTNClicked = true;
     this.limitStatus = null;
   }
 
+  getCheckBoxStatus(): void {
+    const filteredCheckBoxes = this.getCheckBoxInfo().filter((val: BagLimitDto) => {
+      return val.limitIncluded === true;
+    });
+
+    this.areAllCheckBoxEmpty = !filteredCheckBoxes.length;
+  }
+
+  onChecked(id, event): void {
+    const currentBag = this.bags.find((bag) => bag.id === id);
+
+    if (!event.checked) {
+      currentBag.limitIncluded = false;
+    }
+    if (event.checked) {
+      currentBag.limitIncluded = true;
+    }
```

Comment:
remove if operators here. Use currentBag.limitIncluded = event.checked
Also currentBug is local, why do you need it?